services/web3/useCreatePsycSale.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+import { useWaitForTransactionReceipt, useWriteContract } from "wagmi";
+import { psycSaleSepolia } from "@/constants/contracts";
+import psycSaleAbiSepolia from "@/abis/psycSaleAbiSepolia.json";
+
+export const useCreatePsycSale = () => {
+  const { data, writeContract, isPending, error } = useWriteContract();
+
+  const { isLoading: isConfirming, isSuccess: isConfirmed } =
+    useWaitForTransactionReceipt({
+      hash: data
+    });
+
+  const createSale = useCallback(
+    async (
+      tokenIds: number[],
+      saleStartTime: number,
+      floorPrice: number,
+      ceilingPrice: number,
+      merkleRoot: string,
+      ipfsHash: string
+    ) => {
+      return writeContract({
+        address: psycSaleSepolia,
+        functionName: "createSaleBatch",
+        abi: psycSaleAbiSepolia,
+        args: [
+          tokenIds,
+          saleStartTime,
+          floorPrice,
+          ceilingPrice,
```

Comment:
check if we need to send bigInt with eth decimals

Reply:
I will! I'll resolve this today

Reply:
it needed to be bigint
@saksijas I made those changes a while ago